src/tmp_dir.cpp

```diff
@@ -88,9 +88,24 @@ void tmp_dir::execute(const std::string& exe, const std::string& args) const
 
 tmp_dir::~tmp_dir()
 {
+    constexpr int MAX_RETRIES_COUNT = 5;
     if(not enabled(MIGRAPHX_DEBUG_SAVE_TEMP_DIR{}))
     {
-        fs::remove_all(this->path);
+        int count = 0;
+        while(count < MAX_RETRIES_COUNT)
+        {
+            try
+            {
+                fs::remove_all(this->path);
+                break;
+            }
+            catch(const fs::filesystem_error& err)
+            {
+                std::cerr << err.what() << std::endl;
```

Comment:
This should probably say `std::cerr << "Failed to remove " << path << ": " << err.what() << std::endl`.